app/[pokemonName]/components/pokemon-details.jsx

```diff
@@ -0,0 +1,100 @@
+'use client';
+
+import { useState, useEffect } from 'react';
+import Image from 'next/image';
+import Head from 'next/head';
+import styles from './pokemon.module.css';
+import { formatText } from '@/utils/format-text';
+
+export const PokemonPage = ({ pokemonData }) => {
+  const [selectedSprite, setSelectedSprite] = useState('');
+  const [nonNullSprites, setNonNullSprites] = useState([]);
+
+  useEffect(() => {
+    const validSprites = Object.keys(pokemonData.sprites).filter((key) => {
+      const value = pokemonData.sprites[key];
+      return typeof value === 'string' && value.trim() !== '';
+    });
+    setNonNullSprites(validSprites);
+
+    if (validSprites.length > 0) {
+      setSelectedSprite(validSprites[0]);
+    }
+  }, [pokemonData]);
```

Comment:
Since you only need to track sprites, the dependency can be `pokemonData.sprites`